service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -30,6 +39,59 @@ public Claim insertClaim(Claim claim) {
     return claim;
   }
 
+  @Override
+  public AssessmentResult insertAssessmentResult(
+      UUID id,
+      String assessmentResult,
+      String veteranIcn,
+      String claimSubmissionId,
+      String idType,
+      String diagnosticCode)
+      throws NoSuchFieldException {
+    ClaimEntity claimEntity = claimRepository.findById(id).orElse(null);
+    if (claimEntity == null) {
+      throw new NoSuchFieldException("Could not match claim ID {" + id.toString() + "} in DB.");
+    }
+    ObjectMapper mapper = new ObjectMapper();
+    Map<String, Object> evidence = new HashMap<>();
+    try {
+      evidence = mapper.readValue(assessmentResult, Map.class);
+    } catch (Exception e) {
+      log.error("Could not map assessmentResult");
+    }
+    Map summary = (Map) evidence.getOrDefault("evidenceSummary", null);
+    int count = 0;
+    if (summary != null) {
+      for (Object value : summary.values()) {
+        count += (Integer) value;
+      }
+    }
+    AssessmentResultEntity assessmentResultEntity = new AssessmentResultEntity();
+    assessmentResultEntity.setEvidenceCount(count);
+    assessmentResultEntity.setEvidenceCountSummary(summary);
+    for (ContentionEntity contention : claimEntity.getContentions()) {
+      if ((contention.getDiagnosticCode().equals(diagnosticCode))
+          && (contention.getClaim().getId() == id)) {
+        for (AssessmentResultEntity ar : contention.getAssessmentResults())
```

Comment:
These nested structure make my head hurt. Can we simplify a bit by introducing methods? Maybe something like this:

```
var contention = findContention(claim)
var assessmentResult = findAssessmentResult(contention)
...
 
```